src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -0,0 +1,121 @@
+  totalEntries: number = 0;
+  firstLoad = true;
+  user: User;
+  helpLink: string;
+  allEntriesLink: string;
+  public isLoading = true;
+
+  constructor(
+    private registerToolService: RegisterToolService,
+    protected userQuery: UserQuery,
+    protected workflowService: WorkflowService,
+    protected workflowQuery: WorkflowQuery,
+    protected myWorkflowsService: MyWorkflowsService,
+    protected myEntriesQuery: MyEntriesQuery,
+    protected sessionQuery: SessionQuery,
+    protected sessionService: SessionService,
+    private usersService: UsersService
+  ) {
+    super();
+    this.user = this.userQuery.getValue().user;
+  }
+
+  ngOnInit(): void {
+    this.getMyEntries();
+
+    if (this.entryType) {
+      this.entryTypeLowerCase = this.entryType.toLowerCase();
+    }
+
+    //Get the links for the specified entryType
+    if (this.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-workflows.html';
+      this.allEntriesLink = '/my-workflows/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-tools.html';
+      this.allEntriesLink = '/my-tools/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.SERVICE) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/getting-started-with-services.html';
+      this.allEntriesLink = '/my-services/';
+    }
+  }
+
+  getMyEntries() {
+    this.usersService
+      .getUserEntries(null, this.filterText)
+      .pipe(debounceTime(750), takeUntil(this.ngUnsubscribe))
+      .subscribe((myEntries: Array<EntryUpdateTime>) => {
+        this.listOfEntries = [];
+        myEntries.forEach((entry: EntryUpdateTime) => {
+          if (this.entryType === entry.entryType && entry.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
+            if (this.firstLoad) {
+              this.totalEntries += 1;
+            }
+            if (this.listOfEntries.length < 7) {
+              this.listOfEntries.push(entry);
+            }
+          } else if (
+            this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL &&
+            (entry.entryType === EntryUpdateTime.EntryTypeEnum.APPTOOL || entry.entryType === EntryUpdateTime.EntryTypeEnum.TOOL)
+          ) {
+            if (this.firstLoad) {
+              this.totalEntries += 1;
+            }
+            if (this.listOfEntries.length < 7) {
+              this.listOfEntries.push(entry);
+            }
+          } else if (this.entryType === entry.entryType && this.entryType === EntryUpdateTime.EntryTypeEnum.SERVICE) {
+            if (this.firstLoad) {
+              this.totalEntries += 1;
+            }
+            if (this.listOfEntries.length < 7) {
+              this.listOfEntries.push(entry);
+            }
+          }
+        });
+      });
+    this.isLoading = false;
```

Comment:
This needs to be done as part of the subscription, e.g., in a `finalize`. As it is, I see have No Tools Registered briefly displayed because this is done too early.